client/src/app.jsx

```diff
@@ -35,9 +53,15 @@ export default class App extends Component {
           {
             this.state.houseData === null ?
             <p>Hello</p> :
-            <CalendarContainer dates={this.state.dates}/>
+            <CalendarContainer
+            dates={getDates()}
+            clickFn={this.calendarEntryClick.bind(this)}
+            selectedDate={this.state.selectedDate}
```

Comment:
You are proably getting an AirBnB style warning in this area, apperently binding on the passing and declaring state are not best practices. Destructing is preferred above the return { selectedData } = this.state and the bind should be done in in the constructor. The binding could effect performance of your react.